### PR TITLE
Add test step interconnect clusters

### PR DESCRIPTION
**Pull request by electrocucaracha**
This change provides an intermediate step to interconnect worker nodes.

Solves: https://github.com/nephio-project/nephio/issues/270

\cc @johnbelamaric @henderiw

### CODE REVIEW

File: e2e/tests/002.sh [hunk 1]
```diff
@@ -37,3 +37,14 @@ k8s_wait_exists "$kubeconfig" 600 "default" "cluster" "edge02"
 
 k8s_wait_ready "$kubeconfig" 600 "default" "cluster" "edge01"
 k8s_wait_ready "$kubeconfig" 600 "default" "cluster" "edge02"
+
+# Inter-connect worker nodes
+kubectl get secret regional-kubeconfig -o jsonpath='{.data.value}' | base64 -d > $HOME/.kube/regional-kubeconfig
```
Comment (johnbelamaric) [inline e2e/tests/002.sh:42]: I have a [k8s_get_capi_kubeconfig](https://github.com/nephio-project/test-infra/blob/main/e2e/lib/k8s.sh#L108) function that will do this. It uses a tempfile right now but we could instead just put it in .kube.

Non-blocking, this is fine too.

File: e2e/tests/002.sh [hunk 2]
```diff
@@ -37,3 +37,14 @@ k8s_wait_exists "$kubeconfig" 600 "default" "cluster" "edge02"
...
+kubectl get secret edge01-kubeconfig -o jsonpath='{.data.value}' | base64 -d > $HOME/.kube/edge01-kubeconfig
+kubectl get secret edge02-kubeconfig -o jsonpath='{.data.value}' | base64 -d > $HOME/.kube/edge02-kubeconfig
+export KUBECONFIG=$HOME/.kube/config:$HOME/.kube/regional-kubeconfig:$HOME/.kube/edge01-kubeconfig:$HOME/.kube/edge02-kubeconfig
+for context in $(kubectl config get-contexts --no-headers --output name); do
```
Comment (johnbelamaric) [inline e2e/tests/002.sh:46]: This connects the mgmt cluster too. Is that what we want?

File: e2e/provision/README.md [hunk 3]
```diff
@@ -364,6 +364,19 @@ kubectl get secret edge02-kubeconfig -o jsonpath='{.data.value}' | base64 -d > $
 export KUBECONFIG=$HOME/.kube/config:$HOME/.kube/regional-kubeconfig:$HOME/.kube/edge01-kubeconfig:$HOME/.kube/edge02-kubeconfig
 ```
 
+Once the edge clusters are ready it's necessary to inter-connect them. This time
+we are going to use the [containerlab tool](https://containerlab.dev/) for that
+operation.
```
Comment (johnbelamaric) [inline e2e/provision/README.md:369]: Add:

Eventually inter-cluster networking will be automated as well, but it is not yet in this release.

File: e2e/tests/002.sh (hunk 2 above)
Comment (electrocucaracha) [reply e2e/tests/002.sh:46]: The mgmt cluster is an [All-in-One cluster](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/tasks/main.yml#L33-L40), so technically it won't be connected.

**Comment by johnbelamaric** [comment]
I'm running the e2e tests with this branch, if that looks good I will merge it.

**Comment by johnbelamaric** [comment]
Fixes https://github.com/nephio-project/nephio/issues/270

/approve
/lgtm

**Comment by johnbelamaric** [comment]
> Jun 16 00:43:39 nephio-e2e-victor google_metadata_script_runner[1753]: startup-script-url: {"workers":["regional-md-0-qxjm4-6d559bd85dxdq5qp-nfxf8","edge02-md-0-vhqmk-bdd76b47bxdgcpf-9zrdp"]}

Why though is there no edge01 node? Can you look into that @electrocucaracha and submit a follow-up PR?

**Comment by electrocucaracha** [comment]
> > Jun 16 00:43:39 nephio-e2e-victor google_metadata_script_runner[1753]: startup-script-url: {"workers":["regional-md-0-qxjm4-6d559bd85dxdq5qp-nfxf8","edge02-md-0-vhqmk-bdd76b47bxdgcpf-9zrdp"]}
> 
> Why though is there no edge01 node? Can you look into that @electrocucaracha and submit a follow-up PR?

I think that we have to improve the waiting process. I [noticed the edge clusters don't come alive](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/93/e2e/1669714479312539648#1:build-log.txt%3A49382)

**Pull request merged**